libmscore/layout.cpp

```diff
@@ -4624,19 +4645,21 @@ void LayoutContext::collectPage()
         y += cs->height();
     }
 
-    for (int k = 0;; ++k) {
+    //
+    // collect and add systems to the page (until the page is filled or a page break is encountered)
+    //
+    for (;;) {
```

Comment:
where are the exits?
I don't like endless loops where the exits are not within sight

Reply:
I don't like them either, but I didn't create this huge endless loop. I could add more comments but I feel like I've commented sufficiently the entire thing.